### PR TITLE
Deprecate slicing on `_uid`.

### PR DESCRIPTION
**Pull request by jpountz**
`_id` should be used instead on 6.x.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java [hunk 1]
```diff
@@ -91,14 +97,22 @@ public SliceBuilder(String field, int id, int max) {
     }
 
     public SliceBuilder(StreamInput in) throws IOException {
-        this.field = in.readString();
+        String field = in.readString();
+        if (UidFieldMapper.NAME.equals(field) && in.getVersion().before(Version.V_6_3_0)) {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java]: I wonder if this is safe?

Like, if you slice on `_uid` in a mixed version cluster, will this cause you to get duplicate documents and to miss documents?

Comment (nik9000) [reply server/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java]: I see. You've made it safe below. I think that is worth a comment.

Comment (jpountz) [reply server/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java]: Will do.

**Comment by jpountz** [comment]
@nik9000 I added a comment as you suggested.

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**